testplan/common/utils/match.py

```diff
@@ -230,3 +230,97 @@ def match_all(self, regex, timeout=5):
                 raise
 
         return matches
+
+    def match_between(self, regex, mark1, mark2):
```

Comment:
We had got a piece of code like this, but I remembered that it cannot work well on Windows because when reading file in text mode the behavior of ``tell`` is not as expected. Kris had mentioned something:

-----------------------------------
9th Dec 2020 9:20:22 pm Notaisz, Krisztian: 

the problem is that on my windows with my python version tell and seek working with raw file position, while read (when in text mode) reading characters, and a \r\n in windos is just transformed to \n automatically by python
so if I seek just before a newline (on windows)

f.tell()  => x
f.read(1) => '\n'
f.tell() => x+2

so you read one char which is a newline represented by two char in the file which is consumed and file position increased by 2
but again, this is not defined to be like that, you should not have assumption on the result of tell it is just represent a position

https://docs.python.org/3/tutorial/inputoutput.html
 
f.tell() returns an integer giving the file object’s current position in the file represented as number of bytes from the beginning of the file when in binary mode and an opaque number when in text mode. 
well you might have the assumption that positions are comparable as it say opaque numbers
---------------------------------------

So I think that the issue of dealing with '\r\n' is still there, or point to me if I missed something?

Reply:
The previous implementation was try to get a difference of two position, then read that amount of chars. The problem with that was that the difference are in bytes while the read is per char bases which might read 2 bytes in case of '\r\n'

This implementation does not do any difference, it just read from pos1 till the we pass pos2, the only assumption it does that the positions are comparable.